src/EditorFeatures/CSharpTest/UseRecursivePatterns/UseRecursivePatternsTests.cs

```diff
@@ -0,0 +1,1077 @@
+}",
+@"class C
+{
+    bool M(C fieldSymbol)
+    {
+        if (fieldSymbol is { IsConst: true })
+        {
+        }
+    }
+    bool IsConst;
+}");
+        }
+
+        [Fact]
+        public async Task Test21()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        while (expressionType [||]is ArrayTypeSymbol { ElementType: var e1, IsSZArray: var sz1, Rank: var r1 } &&
+               patternType is ArrayTypeSymbol { ElementType: var e2, IsSZArray: var sz2, Rank: var r2 } &&
+               sz1 == sz2 && r1 == r2)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test22()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M(object uncommonData)
+    {
+        if (uncommonData is C deconstruction
+            [||]&& deconstruction.DeconstructMethodInfo.Invocation is BoundCall call)
+        {
+        }
+    }
+    C DeconstructMethodInfo;
+    C Invocation;
+    class BoundCall { }
+}",
+@"class C
+{
+    bool M(object uncommonData)
+    {
+        if (uncommonData is C
+        {
+            DeconstructMethodInfo: { Invocation: BoundCall call },
+        } deconstruction)
+        {
+        }
+    }
+    C DeconstructMethodInfo;
+    C Invocation;
+    class BoundCall { }
+}");
+        }
+
+        [Fact]
+        public async Task Test23()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M(bool a, bool b, bool c)
+    {
+        return
+            this.Kind == 0 &&
+            a &&
+            b &&
+            c [||]&&
+            this.Method == null;
+    }
+    int Kind;
+    object Method;
+}",
+@"class C
+{
+    bool M(bool a, bool b, bool c)
+    {
+        return
+            a && b && c && this is { Kind: 0, Method: null };
+    }
+    int Kind;
+    object Method;
+}");
+        }
+
+        [Fact]
+        public async Task Test24()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        while (initializer.Parent != null &&
+               initializer.Parent.Kind == SyntaxKind.SimpleAssignmentExpression &&
+               ((AssignmentExpressionSyntax)initializer.Parent).Right == initializer &&
+               initializer.Parent.Parent != null [||]&&
+               initializer.Parent.Parent.Kind == SyntaxKind.ObjectInitializerExpression)
+        {
+        }
+    }
+    class AssignmentExpressionSyntax : C { }
+    C Right;
+    C Parent;
+    SyntaxKind Kind;
+    const C initializer = null;
+    enum SyntaxKind { SimpleAssignmentExpression, ObjectInitializerExpression }
+}",
+@"class C
+{
+    bool M()
+    {
+        while (initializer is
+        {
+            Parent: AssignmentExpressionSyntax
+            {
+                Kind: SyntaxKind.SimpleAssignmentExpression,
+                Right: initializer,
+                Parent: { Kind: SyntaxKind.ObjectInitializerExpression },
+            },
+        })
```

Comment:
we really need to fix indentation here

Reply:
I only wrap property patterns, the rest is on the default formatter. I think the fix should be there.